test/issue-117-format-error.cpp

```diff
@@ -0,0 +1,69 @@
+// issue-00-format-error.cpp
+
+#include "nlohmann/json-schema.hpp"
+#include "nlohmann/json.hpp"
+#include <iostream>
+
+static int error_count = 0;
+
+#define CHECK_THROW(x, msg)                \
+	{                                      \
+		bool fail = false;                 \
+		try {                              \
+			x;                             \
+		} catch (std::exception &) {       \
+			fail = true;                   \
+		}                                  \
+		if (fail == false) {               \
+			++error_count;                 \
+			std::cout << msg << std::endl; \
+		}                                  \
+	}
+
+#define CHECK_NO_THROW(x, msg)                                        \
+	{                                                                 \
+		bool fail = false;                                            \
+		std::string exception_error;                                  \
+		try {                                                         \
+			x;                                                        \
+		} catch (std::exception & e) {                                \
+			fail = true;                                              \
+			exception_error = e.what();                               \
+		}                                                             \
+		if (fail == true) {                                           \
+			++error_count;                                            \
+			std::cout << msg << ": " << exception_error << std::endl; \
+		}                                                             \
+	}
```

Comment:
We should share this kind of defines between all the tests.

Reply:
sorry, I didn't understand you

Reply:
I mean, these macros could go into a seperate .h-file and be used by others tests.

Reply:
Yes, of course, but maybe better use some complete test library? For example Catch2

Reply:
Yes, you're right, I never used it, but it seems to be time to use it here.